convert2rhel/checks.py

```diff
@@ -449,3 +453,47 @@ def _bad_kernel_substring(kernel_release):
         )
         return True
     return False
+
+
+def check_package_updates():
+    """Ensures that the system packages installed are up-to-date."""
+    logger.task("Prepare: Checking if the packages are up-to-date")
+
+    yum_stdout, _ = run_subprocess("yum check-update -q", print_output=False)
+
+    # There is a whole blank line before outputting the packages to update
+    # and to output the real counting, we just -1 the counting.
+    packages_to_update_count = yum_stdout.count("\n") - 1
+
+    if packages_to_update_count > 0:
+        logger.critical(
+            "The system has {0} packages to be updated. "
+            "To proceed with the conversion, update the packages on your system by executing the following step:\n\n"
+            "1. Run: yum update -y\n".format(packages_to_update_count)
+        )
+    else:
+        logger.info("System is up-to-date.")
+
+
+def check_live_kernel_version():
+    """Check if the installed kernel is behind or at the same version in yum repos"""
+    logger.task("Prepare: Checking if the installed kernel version is the most recent.")
+
+    # The latest kernel version on repo
+    repoquery_output, _ = run_subprocess('repoquery --qf "%{version}-%{release}.%{arch}" kernel')
+
+    # The installed kernel version
+    uname_output, _ = run_subprocess("uname -r")
+
+    match = LooseVersion(repoquery_output) > LooseVersion(uname_output)
```

Comment:
I see that we `import rpm` in the pkghandler file so we already have a dependency on that.  rpm has a function to compare NEVR (Name epoch version release). `rpm.labelCompare((epoch1, version1, release1), (epoch2, version2, release2))`.  

convert2rhel is GPLv3 or later and rpmdevtools is GPLv2 or later (correct me if I'm wrong on either of those points :-) so you should be able to look at the source of rpmdev-rpmvercmp for some additional examples: https://pagure.io/rpmdevtools/blob/main/f/rpmdev-vercmp#_108

Reply:
Note that arch probably doesn't make sense as part of greater than and less than comparisons.  But it would make sense for an equality comparison.  Perhaps you want to do something to make sure that you have the right arch prior to doing your version comparison? (Or you might want to assume correct arch due to having setup the repositories earlier?)

Reply:
Do you guys think that comparing versions is a really important step on this? I mean, `repoquery` should output the latest version available at the current repositories, so if the `uname -r` doesn't match the output of `repoquery` it would mean that it's not updated, right? 

Don't know if I'm really on the correct train of thought here. If not, I can change to do version comparison with `rpm.labelCompare()`

Reply:
I'm not familiar enough to give a definitive answer but here are some ideas of potential problems:

* What do we want to do if the repository is out-of-sync and reports an older version?
* What do we want to do in this function if the user has a non-standard but newer kernel installed?

Reply:
In a recent call with @bocekm we were able to get this right. I've added the code for properly handling the `repoquery` and `uname` output, and to add the `rpm.labelCompare` function too.